asset/src/job_metric_api/api.ts

```diff
@@ -0,0 +1,208 @@
+        }
+    }
+
+    /**
+     * [decrement metric by value]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value, default = 1]
+     * @return {Promise<void>}
+     */
+    async dec(name: string, labelValues: Array<string>, value = 1): Promise<void> {
+        if (this.metricList[name].functions.has('dec')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).dec(value);
+        }
+    }
+    /**
+     * [observe value, used by summary metric type]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value]
+     * @return {Promise<void>}
+     */
+    async observe(name: string, labelValues: Array<string>, value: number): Promise<void> {
+        if (this.metricList[name].functions.has('observe')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).observe(value);
+        }
+    }
+
+    /**
+     * [add (define) new metric]
+     * @param  {string} name [metric name]
+     * @param  {string} help [metric help]
+     * @param  {Array<string>} labelsNames [list of label names]
+     * @param  {string} type [metric type, gauge, counter, histogram, or summary]
+     * @return {Promise<void>}
+     */
+    async add(name: string, help: string, labelsNames: Array<string>,
+        type: string, buckets: Array<number> = [0.1, 5, 15, 50, 100, 500]): Promise<void> {
+        if (!(name in this.metricList)) {
+            const fullname = this.prefix + name;
+            if (type === 'gauge') {
+                this.metricList[name] = await this._createGaugeMetric(fullname, help,
+                    labelsNames.concat(Object.keys(this.default_labels)));
+            }
+            if (type === 'counter') {
+                this.metricList[name] = await this._createCounterMetric(fullname, help,
+                    labelsNames.concat(Object.keys(this.default_labels)));
+            }
+            if (type === 'histogram') {
+                this.metricList[name] = await this._createHistogramMetric(fullname, help,
+                    labelsNames.concat(Object.keys(this.default_labels)), buckets);
+            }
+        }
+    }
+    /**
+     * [addSummary (define) new summary metric]
+     * @param  {string} name [metric name]
+     * @param  {string} help [metric help]
+     * @param  {Array<string>} labelsNames [list of label names]
+     * @param  {Array<number>} percentiles [metric percentiles, default[0.01, 0.1, 0.9, 0.99] ]
+     * @param  {number} maxAgeSeconds [how old a bucket can be before it is reset ]
+     * @param  {number} ageBuckets [how many buckets for sliding window ]
+     * @return {Promise<void>}
+     */
+    async addSummary(name: string,
```

Comment:
The name of this function should be `addSummaryMetric` since it more about adding a metric rather than changing a value of a summary bucket